### PR TITLE
#43 - Create packetizer

### PR DESCRIPTION
**Pull request by Glamhoth**

### CODE REVIEW

File: src/Packetizer/Packetizer.h [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+/// @brief  Possible Space Packet types
+typedef enum {
+    Packetizer_PacketType_Telemetry = 0,  ///< Telemetry packet
+    Packetizer_PacketType_Telecommand = 1 ///< Telecommand packet
+} Packetizer_PacketType;
+
+/// @brief  Enumeration listing possible error codes.
+typedef enum {
+    Packetizer_ErrorCode_IncorrectCrc16 = 1,
```
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/Packetizer/Packetizer.h:45]: Done

File: src/Packetizer/SpacePacket.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SPACE_PACKET_H
+#define SPACE_PACKET_H
+
+#define SPACE_PACKET_PRIMARY_HEADER_SIZE_BYTES 6u
```
Comment (rphh) [inline src/Packetizer/SpacePacket.h:26]: This shall be public. Some other constants (which are checked) too.

File: README.md [hunk 3]
```diff
@@ -1 +1,17 @@
-# TASTE-Runtime-Common
\ No newline at end of file
+# TASTE-Runtime-Common
+
```
[the remark this answers is not in the document]
Comment (Glamhoth) [reply README.md:2]: Added.

File: src/Packetizer/Packetizer.c [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                     const Packetizer_PacketType packetType,
+                     const uint16_t source,
+                     const uint16_t destination,
+                     uint8_t* const dataPointer,
+                     const size_t dataOffset,
+                     const size_t dataSize)
+{
+    // Unused in this implementation
+    (void)source;
+    (void)dataOffset;
```
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/Packetizer/Packetizer.c:44]: It doesn't actually. Removed that.

File: src/Packetizer/Packetizer.c [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+                       uint16_t* const destination,
+                       size_t* const dataOffset,
+                       size_t* const dataSize,
+                       uint32_t* const errorCode)
+{
+    // Unused in this implementation
+    (void)self;
+    (void)source;
+
+    (void)packetSize;
```
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/Packetizer/Packetizer.c:84]: Removed.

File: src/Packetizer/Packetizer.c [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+    assert(packetPointer != NULL);
+    assert(packetSize > SPACE_PACKET_PRIMARY_HEADER_SIZE_BYTES + SPACE_PACKET_ERROR_CONTROL_SIZE_BYTES);
+    assert(dataOffset != NULL);
+    assert(dataSize != NULL);
+    assert(destination != NULL);
+
+    uint16_t receivedCrc = packetPointer[packetSize - 1] | (packetPointer[packetSize - 2] << 8);
```
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/Packetizer/Packetizer.c:92]: Done

File: src/Packetizer/Packetizer.c [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+            *errorCode = Packetizer_ErrorCode_IncorrectPacketType;
+        }
+        return false;
+    }
+
+    // Save the results
+    *destination = packetPointer[1]
+                   | (packetPointer[0] & SPACE_PACKET_APID_HIGH_BITS_MASK) << (8u - SPACE_PACKET_APID_HIGH_BITS_OFFSET);
+    *dataOffset = SPACE_PACKET_PRIMARY_HEADER_SIZE_BYTES;
+    *dataSize = (size_t)(packetPointer[4] << 8u) | packetPointer[5];
```
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/Packetizer/Packetizer.c:116]: Done

File: src/Packetizer/Packetizer.h [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+ */
+bool Packetizer_depacketize(const Packetizer* const self,
+                            const Packetizer_PacketType packetType,
+                            const uint8_t* const packetPointer,
+                            const size_t packetSize,
+                            uint16_t* const source,
+                            uint16_t* const destination,
+                            size_t* const dataOffset,
+                            size_t* const dataSize,
+                            uint32_t* const errorCode);
```
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/Packetizer/Packetizer.h:102]: That's right, changed

File: src/Packetizer/Packetizer.h [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+ * @param[in]       dataOffset  Offset to where data starts
+ * @param[in]       dataSize    Size of the data
+ *
+ * @returns Packet size.
+ */
+size_t Packetizer_packetize(Packetizer* const self,
+                            const Packetizer_PacketType packetType,
+                            const uint16_t source,
+                            const uint16_t destination,
+                            uint8_t* const dataPointer,
```
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/Packetizer/Packetizer.h:75]: Sure, changed

File: src/Packetizer/SpacePacket.h (hunk 2 above)
Comment (Glamhoth) [reply src/Packetizer/SpacePacket.h:26]: Done, SpacePacket.h is now a public header + created SpacePacketInternal.h for masks and offsets.

Comment (rphh) [reply src/Packetizer/SpacePacket.h]: Good

**Review by Lurkerpas — APPROVED** [review]

**Pull request merged**